code/datums/antagonists/datum_vampire.dm

```diff
@@ -0,0 +1,280 @@
+	if(!iscarbon(owner))
+		H.LAssailant = null
+	else
+		H.LAssailant = O
+	playsound(O.loc, 'sound/weapons/bite.ogg', 50, 1)
+	while(do_mob(O, H, 50))
+		if(!is_vampire(O))
+			to_chat(O, "<span class='warning'>Your fangs have disappeared!</span>")
+			return
+		old_bloodtotal = total_blood
+		old_bloodusable = usable_blood
+		if(!H.blood_volume)
+			to_chat(O, "<span class='warning'>They've got no blood left to give.</span>")
+			break
+		if(H.stat != DEAD)
+			blood = min(20, H.blood_volume)// if they have less than 20 blood, give them the remnant else they get 20 blood
+			total_blood += blood / 2	//divide by 2 to counted the double suction since removing cloneloss -Melandor0
+			usable_blood += blood / 2
+		else
+			blood = min(5, H.blood_volume)	// The dead only give 5 blood
+			total_blood += blood
+		check_vampire_upgrade()
+		if(old_bloodtotal != total_blood)
+			to_chat(O, "<span class='notice'><b>You have accumulated [total_blood] [total_blood > 1 ? "units" : "unit"] of blood[usable_blood != old_bloodusable ? ", and have [usable_blood] left to use" : ""].</b></span>")
+		H.blood_volume = max(H.blood_volume - 25, 0)
+		if(ishuman(O))
+			O.nutrition = min(O.nutrition + (blood / 2), NUTRITION_LEVEL_WELL_FED)
+		playsound(O.loc, 'sound/items/eatfood.ogg', 40, 1)
+
+	draining = null
+	to_chat(owner, "<span class='notice'>You stop draining [H.name] of blood.</span>")
+
+/datum/antagonist/vampire/proc/force_add_ability(path)
+	var/spell = new path(owner)
+	if(istype(spell, /obj/effect/proc_holder/spell))
+		owner.AddSpell(spell)
+	powers += spell
+
+/datum/antagonist/vampire/proc/get_ability(path)
+	for(var/P in powers)
+		var/datum/power = P
+		if(power.type == path)
+			return power
+	return null
+
+/datum/antagonist/vampire/proc/add_ability(path)
+	if(!get_ability(path))
+		force_add_ability(path)
+
+/datum/antagonist/vampire/proc/remove_ability(ability)
+	if(ability && (ability in powers))
+		powers -= ability
+		owner.spell_list.Remove(ability)
+		qdel(ability)
+
+
+/datum/antagonist/vampire/proc/remove_vampire_powers()
+	for(var/P in powers)
+		remove_ability(P)
+	owner.current.alpha = 255
+
+/datum/antagonist/vampire/proc/check_vampire_upgrade(var/announce = TRUE)
+	var/list/old_powers = powers.Copy()
+	for(var/ptype in upgrade_tiers)
+		var/level = upgrade_tiers[ptype]
+		if(total_blood >= level)
+			add_ability(ptype)
+	if(announce)
+		announce_new_power(old_powers)
+	owner.current.update_sight() //deal with sight abilities
+
+/datum/antagonist/vampire/proc/announce_new_power(list/old_powers)
+	for(var/p in powers)
+		if(!(p in old_powers))
+			if(istype(p, /obj/effect/proc_holder/spell))
+				var/obj/effect/proc_holder/spell/power = p
+				to_chat(owner.current, "<span class='notice'>[power.gain_desc]</span>")
+			else if(istype(p, /datum/vampire_passive))
+				var/datum/vampire_passive/power = p
+				to_chat(owner, "<span class='notice'>[power.gain_desc]</span>")

```

Comment:
why would the upper one be owner.current but this one just owner? make up your mind.